### ISSUE TITLE
Need `treesaver.ready` or equivalent event

### ISSUE THREAD

**Issue by fortes** [Treesaver/treesaver#210]
Make it easier for developers to do actions once content is loaded.

**Comment by bramstein** [comment]
There is the `treesaver.index.loaded` event, which fires when the index has finishing loading and parsing, but perhaps we want some event before then.

**Comment by fortes** [comment]
I'm thinking perhaps it should be fired when the user is able to interact / view content. Could be used by developers to show toolbar/lightbox or other such things.

articlechanged/pagechanged events are currently too noisy, but perhaps once fixed we could rely on those instead of creating a new event.

**Comment by mithun12000** [comment]
i have try using it but not working.

treesaver.addListener(document, 'treesaver.index.loaded', function (event) {
  var index = event.index,
      one = document.getElementsByClassName('one')[0],
      two = document.getElementsByClassName('two')[0],
      three = document.getElementsByClassName('three')[0],
      dynamic = document.getElementsByClassName('dynamic')[0],
      counter = 4;
}

here treesaver.index.loaded never fire.

also event is undefined.

can you tell me what should i use instead of this?

**Comment by bramstein** [comment]
Are you using the latest build? This only works with Treesaver 0.9.3-beta.

On Thu, Jul 21, 2011 at 2:19 PM, mithun12000
reply@reply.github.com
wrote:

**Comment by mithun12000** [comment]
No I am using stable one 0.9.2

will it not work on it?

**Comment by mithun12000** [comment]
how can i dont 0.9.3 . also can i have full documentation for this framework?

**Comment by bramstein** [comment]
It won't work on 0.9.2 as the table of contents handling is completely
changed in 0.9.3. You can download a copy here:

https://github.com/downloads/bramstein/treesaver/treesaver-0.9.3-beta.js

It is still beta, so there might be some problems. We haven't updated
the documentation to reflect the changes in 0.9.3 yet as it hasn't
been officially released. There is a draft of the documentation for
the new table of contents: https://gist.github.com/946119 There are
also several examples for 0.9.3 in the repository:

https://github.com/Treesaver/treesaver/tree/master/examples

These only work with 0.9.3.

For 0.9.2 the best resource is still the Wiki:
https://github.com/Treesaver/treesaver/wiki

On Fri, Jul 22, 2011 at 9:52 AM, mithun12000
reply@reply.github.com
wrote:

**Comment by mithun12000** [comment]
https://github.com/downloads/bramstein/treesaver/treesaver-0.9.3-beta.js this is a JS min (compressed) edition.  can you put a uncompressed edition as well so that i can modify code if need as well as understand properly as there is no documentation. I cant build this as there is some problem in py script which says that git is not an internal command (windows).

will go through those url now.

thank you.

**Comment by mithun12000** [comment]
also

{{thumb}}
{{url}}

this line dont parse properly.

**Comment by bramstein** [comment]
The source code is in the the main repository. We do not have support
for building on Windows unfortunately, but I built a debug version for
you:

http://cloud.github.com/downloads/bramstein/treesaver/treesaver-0.9.3-beta.debug.js

On Fri, Jul 22, 2011 at 11:29 AM, mithun12000
reply@reply.github.com
wrote:

**Comment by bramstein** [comment]
The template variables only get expanded when you have a corresponding
property in your JSON index file. If you don't always have those keys
you can use a conditional, for example:

{{#title}}
  The title is: {{title}}
{{/title}}

This will only print out the title if it is available. You can find
more details on the templating language at:
http://mustache.github.com/mustache.5.html

On Fri, Jul 22, 2011 at 11:31 AM, mithun12000
reply@reply.github.com
wrote:

**Comment by mithun12000** [comment]
thanks for building a debug version for me. 

i will check mustache now.

**Comment by mithun12000** [comment]
Hi guys,

i was facing one problem which is due to "querySelectorAll", this give all attribute value as escape string, due to this all template value ({{tag}}) is escape char then it was not getting replace with variable. 

In order to fix this i tried something and it worked in my case. let me know if that code is ok or not.

to_html: function(template, view, partials, send_fun) {  
      template = unescape(template); //this code is added
      var renderer = new Renderer();
      if(send_fun) {
        renderer.send = send_fun;
      }
      renderer.render(template, view, partials);
      if(!send_fun) {
        return renderer.buffer.join("\n");
      }
}

under class name Renderer or may be under Mustache.

regards
mithun

**Comment by bramstein** [comment]
Good catch, I've noticed problems with the escaping as well. I'm considering switching Treesaver to handlebars.js, which features the same syntax as mustache.js, but allows you to define custom escaping functions.
